core/commonMain/src/implementations/immutableMap/PersistentHashMap.kt

```diff
@@ -104,6 +104,12 @@ internal class PersistentHashMap<K, V>(internal val node: TrieNode<K, V>,
         }
     }
 
+    /**
+     * We provide [equals], so as a matter of style, we should also provide [hashCode].
+     * However, the implementation from [AbstractMap] is enough.
+     */
+    override fun hashCode(): Int = super<AbstractMap>.hashCode()
```

Comment:
Can we add a similar `hashCode` implementation to the affected builders as well?
Namely, `PersistentHashMapBuilder`, `PersistentOrderedMapBuilder`, `PersistentOrderedSetBuilder`.